Bundle/Model/Row/Subscribers.php

```diff
@@ -89,4 +89,28 @@ public function writeLog($message, $state = null, $saveImmediatly = false)
         ));
         if ($saveImmediatly) $childRow->save();
     }
+
+    protected function _beforeDelete()
+    {
+        parent::_beforeDelete();
+
+        $select = new \Kwf_Model_Select();
+        $select->whereEquals('subscriber_id', $this->id);
+        $this->getModel()->getDependentModel('Logs')->deleteRows($select);
+        $this->getModel()->getDependentModel('ToCategories')->deleteRows($select);
+    }
+
+    public function deleteAndHash()
+    {
+        $model = \Kwf_Model_Abstract::getInstance('KwcNewsletter\Bundle\Model\DeletedSubscriberHashes');
+
+        $hash = \Kwf_Util_Hash::hash($this->email);
```

Comment:
bitte mach nicht so einen guten hash (incl salt usw) das macht es eventuell unnötig schwer zukünftig mit den daten zu arbeiten.

mein vorschlag: sha1($this->email)

Reply:
@nsams ok ist gefixed